### PR TITLE
Feature: Applet list api / Interceptors / Secure Storage

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/entities/user/model/user.slice.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  fullName: null,
+  id: null,
+}
+
+const userSlice = createSlice({
+  name: "user",
+  initialState,
+  reducers: {
+    save: (state, action: PayloadAction<UserStore>) => {
+      return { ...state, ...action.payload }
```
Comment (BamMironov) [inline src/entities/user/model/user.slice.ts:20]: Redux Toolkit uses Immer internally
https://redux-toolkit.js.org/usage/immer-reducers

please make use of it when writing reducers.

File: src/features/ChangePassword/ui/ChangePasswordForm.tsx [hunk 2]
```diff
@@ -61,6 +61,8 @@ export const ChangePasswordForm = ({ title, token, email, onSuccessExtended }: C
   const approveErrorMessage = useMemo(() => approveError?.evaluatedMessage, [approveError])
   const updateErrorMessage = useMemo(() => updateError?.evaluatedMessage, [updateError])
 
+  const approveSuccessMessage = useMemo(() => (isApproveSuccess ? t("success") : null), [isApproveSuccess, t])
```
Comment (BamMironov) [inline src/features/ChangePassword/ui/ChangePasswordForm.tsx:64]: I didn't notice that during previous reviews but anyway: You don't need `useMemo` here. If it is a dead simple computation then we don't wrap it in useMemo

File: src/features/ProtectedRoute/ui/InactivityTracker.tsx [hunk 3]
```diff
@@ -2,36 +2,40 @@ import { PropsWithChildren, useCallback, useEffect, useRef } from "react"
...
   const navigate = useNavigate()
+  const { clearUser } = userModel.hooks.useUserState()
 
-  const { clearUserAndAuth, auth } = useAuth()
-  const { mutate: logout } = useLogoutMutation()
+  const { mutate: logout } = useLogoutMutation({
+    onSuccess() {
+      clearUser()
+      userModel.secureTokensStorage.clearTokens()
+      navigate(ROUTES.login.path)
```
Comment (BamMironov) [inline src/features/ProtectedRoute/ui/InactivityTracker.tsx:25]: All of these need to be performed even if the logout request fails

File: src/features/Signup/ui/SignupForm.tsx [hunk 4]
```diff
@@ -21,10 +23,20 @@ export const SignupForm = () => {
...
 
-  const { mutate: signup, error, isSuccess } = useSignupMutation()
+  const {
+    mutate: signup,
+    error,
+    isSuccess,
+    isLoading,
+  } = useSignupMutation({
+    onSuccess() {
+      reset()
```
Comment (BamMironov) [inline src/features/Signup/ui/SignupForm.tsx:37]: Why?

File: src/shared/api/services/axios.ts [hunk 5]
```diff
@@ -7,4 +14,50 @@ const axiosService = axios.create({
...
+      }
+
+      try {
+        const { data } = await authorizationService.refreshToken({ refreshToken: tokens?.refreshToken })
+
+        secureTokensStorage.setTokens(data.result)
+
+        config.headers!.Authorization = `${data.result.tokenType} ${data.result.accessToken}`
+      } catch (e) {
+        Promise.reject(e)
```
Comment (BamMironov) [inline src/shared/api/services/axios.ts:53]: What happens if refresh token is expired?

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/entities/user/model/user.slice.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/entities/user/model/user.slice.ts:20]: Done

File: src/features/ChangePassword/ui/ChangePasswordForm.tsx (hunk 2 above)
Comment (moiskillnadne) [reply src/features/ChangePassword/ui/ChangePasswordForm.tsx:64]: Done

File: src/features/Signup/ui/SignupForm.tsx (hunk 4 above)
Comment (moiskillnadne) [reply src/features/Signup/ui/SignupForm.tsx:37]: Done

File: src/shared/api/services/axios.ts (hunk 5 above)
Comment (moiskillnadne) [reply src/shared/api/services/axios.ts:54]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**